java/runtime/src/main/java/org/ray/runtime/functionmanager/FunctionManager.java

```diff
@@ -92,31 +100,37 @@ public RayFunction getFunction(JobId jobId, RayFunc func) {
    * @param functionDescriptor The function descriptor.
    * @return A RayFunction object.
    */
-  public RayFunction getFunction(JobId jobId, JavaFunctionDescriptor functionDescriptor) {
+  public RayFunction getFunction(JobId jobId,
+      JavaFunctionDescriptor functionDescriptor) {
     JobFunctionTable jobFunctionTable = jobFunctionTables.get(jobId);
     if (jobFunctionTable == null) {
-      ClassLoader classLoader;
-      if (Strings.isNullOrEmpty(jobResourcePath)) {
-        classLoader = getClass().getClassLoader();
-      } else {
-        File resourceDir = new File(jobResourcePath + "/" + jobId.toString() + "/");
-        Collection<File> files = FileUtils.listFiles(resourceDir,
-            new RegexFileFilter(".*\\.jar"), DirectoryFileFilter.DIRECTORY);
-        files.add(resourceDir);
-        final List<URL> urlList = files.stream().map(file -> {
-          try {
-            return file.toURI().toURL();
-          } catch (MalformedURLException e) {
-            throw new RuntimeException(e);
+      synchronized (jobFunctionTablesLock) {
+        jobFunctionTable = jobFunctionTables.get(jobId);
+        if (jobFunctionTable == null) {
```

Comment:
maybe wrap following code in a method (e.g., `createJobFunctionTable`) so the structure is clearer?

Reply:
Done.